src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -1029,7 +1029,27 @@ public void onShowProfiler(OpenProfileEvent event)
       
       // create new profiler 
       ensureVisible(true);
-      server_.newDocument(
+
+      if (event.getDocId() != null)
+      {
+         server_.getSourceDocument(event.getDocId(), new ServerRequestCallback<SourceDocument>()
+         {
+            @Override
+            public void onResponseReceived(SourceDocument response)
+            {
+               addTab(response);
+            }
+            
+            @Override
+            public void onError(ServerError error)
+            {
+               Debug.logError(error);
```

Comment:
Is `OpenProfileEvent` fired as a result of a user operation? If it is, we should give the user some feedback if this is where control flow ends.